hub/core/chunk.py

```diff
@@ -72,6 +118,39 @@ def is_under_min_space(self, min_data_bytes_target: int) -> bool:
     def has_space_for(self, num_bytes: int, max_data_bytes: int):
         return self.num_data_bytes + num_bytes <= max_data_bytes
 
+    def extend_samples(
+        self,
+        buffer: memoryview,
+        max_data_bytes: int,
+        shapes: Sequence[Tuple[int]],
+        nbytes: Sequence[int],
+    ):
+        """Store `buffer` in this chunk.
+        Args:
+            buffer (memoryview): Buffer that represents multiple samples of same shape
+            max_data_bytes (int): Used to determine if this chunk has space for `buffer`.
+            shapes (Sequence[Tuple[int]]): Shape for each sample
+            nbytes (Sequence[int]): Number of bytes in each sample
+        Raises:
+            FullChunkError: If `buffer` is too large.
+        """
+        incoming_num_bytes = len(buffer)
+
+        if not self.has_space_for(incoming_num_bytes, max_data_bytes):
+            raise FullChunkError(
+                f"Chunk does not have space for the incoming bytes (incoming={incoming_num_bytes}, max={max_data_bytes})."
+            )
+
+        # `_data` will be a `memoryview` if `frombuffer` is called.
+        if isinstance(self._data, memoryview):
+            self._data = bytearray(self._data)
```

Comment:
```suggestion
        self._make_data_bytearray()
```

this is already a class function (replace all instances of this copy-pasted block elsewhere too)